### DISCUSSION TITLE
Use just for sensor data

### DISCUSSION THREAD

**Discussion by martenjacobs** [lbbrhzn/ocpp#231] [Q&A] [answered]
I've just installed by Wallbox a few days ago and thought I'd use this component to monitor it without depending on the Wallbox cloud component (I'm specifically interested in the power consumption because it's on a separate circuit than my house).
It connected fine, but after plugging in the car last night, it didn't start charging.
After a few tries, I started flicking some switches in Home Assistant and it started to charge. 
I think what I did was enable the "Charge Control" switch. Then, after the car was done charging, that was switched off again.
I'm guessing this is by design, but it doesn't really work for my use case. I just want to be able to plug in my car at any time and it should be allowed to start charging. Is there some configuration I can do to get it to work like that?

**Comment by drc38** [comment]
I suggest trying again, as provided the charger status is "Available" nothing should prevent the car charging in HA when plugged in (if you have some sort of charge timer on the car enabled that will however).

**Reply by martenjacobs** [reply]
OK, it's good to know that by default it should just work. I'm seeing that the "Available" switch is off right now, and when I switch it on, it turns itself off again. Has that got something to do with the fact that the car is plugged in (and fully charged) right now? The status is `SuspendedEV`
On a (sort-of) side note: is there any documentation about the different switches and sensors?

**Reply by drc38** [reply]
If you read the ocpp 1.6 specification the switches and statuses will make more sense. https://www.openchargealliance.org/

**Reply by lbbrhzn** [reply]
Hi @martenjacobs,

My wallbox also does not automatically start charging. I have an automation that turns on the Charge switch when the status is 'preparing'. I think that status of 'suspendedEV' means that the EV has suspended charging, for instance because it has reached the target state of charge.

**Reply by martenjacobs** [reply]
@lbbrhzn could you share the automation?

**Reply by lbbrhzn** [reply]
alias: Start charging
description: Start charging when status is preparing
trigger:
  - platform: state
    entity_id: sensor.charger_status
    to: Preparing
    for:
      hours: 0
      minutes: 0
      seconds: 10
      milliseconds: 0
condition: []
action:
  - type: turn_on
    device_id: 14badec809585457dcac7e2f61080d75
    entity_id: switch.charger_charge_control
    domain: switch
mode: single

**Reply by drc38** [reply]
I suggest checking the value of this configuration key. Extract from ocpp specification as follows "If  the  value  of  AuthorizeRemoteTxRequests  is  false,  the  Charge  Point  SHALL  immediately  try  to  start  a transaction  for  the  idTag  given  in  the  RemoteStartTransaction.req  message."

**Comment by drc38** [comment]
Another key to check "AllowOfflineTxForUnknownId" which should be set to True in case there is a comms issue between HA and the charger when you plug in the car.

**Reply by drc38** [reply]
This charger automatically starts charging when plugged in, so you can compare config values https://docs.evnex.io/docs/ex-series-ocpp-configuration-settings

**Comment by martenjacobs** [comment, marked as answer]
Just to follow up:
I compared the configuration settings of my Wallbox Pulsar Plus with the list provided by @drc38. I didn't notice anything out of the ordinary there. I subsequently found [this document](https://support.wallbox.com/wp-content/uploads/2021/02/EN_OCPP_ACTIVATION_GUIDE.pdf), which states that Wallboxes disable "Anonymous charging sessions" when connected to OCPP. I'm guessing this is what I'm experiencing.
I'm now going to use [the automation provided](https://github.com/lbbrhzn/ocpp/discussions/231#discussioncomment-1793752) by @lbbrhzn, which seems to work well.
Unfortunately, I can't mark that reply as the answer.

**Reply by drc38** [reply]
Interesting to know, I see my charger has a custom configuration key that can be set to enable or disable automatic charging. It could be worth checking yours in case there is something similar using the get configuration service.

**Reply by lbbrhzn** [reply]
@martenjacobs,  I've also got a Wallbox Pulsar Plus, thanks for the document! So far I've found the automation works reliably.

**Comment by srirams** [comment]
According to the wallbox docs, setting "AuthEnabled" to false should allow the charger to charge without requiring authorization.

```
action: ocpp.configure
data:
  devid: "charger"
  ocpp_key: "AuthEnabledOffline"
  value: false
```

and for good measure

```
action: ocpp.configure
data:
  devid: "charger"
  ocpp_key: "AuthEnabled"
  value: false
```

and setting this works on my wallbox pulsar. I don't know if these config changes survive a reboot or reconnect.
